vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java

```diff
@@ -132,6 +140,65 @@ public String toString()
         });
     }
 
+    /**
+     * Waits for an <b><i>element</i></b> with the specified <b>locator</b> to stop moving
+     * @param locator locator to locate element
+     */
+    @When("I wait until element located by `$locator` stops moving")
+    public void waitForElementStopsMoving(Locator locator)
+    {
+        uiContext.getOptionalSearchContext().ifPresent(context -> {
+            Optional<WebElement> webElement = baseValidations.assertElementExists("The element to validate stop moving",
+                    context, locator);
+            locator.getSearchParameters().setWaitForElement(false);
+            IExpectedSearchContextCondition<Boolean> condition = new IExpectedSearchContextCondition<>() {
+                private Point lastPosition;
+                private boolean elementPresent = true;
+                private Optional<WebElement> optionalWebElement = webElement;
+
+                @Override
+                public Boolean apply(SearchContext context)
+                {
+                    try
+                    {
+                        return optionalWebElement.map(WebElement::getLocation).map(currentPosition ->
+                        {
+                            boolean samePosition = currentPosition.equals(lastPosition);
+                            if (!samePosition)
+                            {
+                                LOGGER.info("The element is moved at '{}'", currentPosition);
+                            }
+                            lastPosition = currentPosition;
+                            elementPresent = true;
```

Comment:
why is it needed to set the value at this point?

Reply:
Now it was set to _true_ for StaleElementReferenceException, which follows not exactly after the previous one.
@valfirst, maybe we don't need to store  _elementPresent_ variable at all if we're trying to find the element in  _catch_ block?